internal/trace/ot/ot.go

```diff
@@ -0,0 +1,128 @@
+// Package ot wraps github.com/opentracing/opentracing-go and
+// github.com./opentracing-contrib/go-stdlib with selective tracing behavior that is toggled on and
+// off with the presence of a context item (uses context.Context). This context item is propagated
+// across API boundaries through a HTTP header (X-Sourcegraph-Should-Trace).
+package ot
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+
+	"github.com/opentracing-contrib/go-stdlib/nethttp"
+	"github.com/opentracing/opentracing-go"
+)
+
+type tracePolicy string
+
+const (
+	// TraceNone turns off tracing.
+	TraceNone tracePolicy = "none"
+
+	// TraceSelective turns on tracing only for requests with the X-Sourcegraph-Should-Trace header
+	// set to a truthy value.
+	TraceSelective = "selective"
+
+	// Comprehensive turns on tracing for all requests.
+	TraceAll = "all"
```

Comment:
```suggestion
	TraceSelective tracePolicy = "selective"

	// Comprehensive turns on tracing for all requests.
	TraceAll tracePolicy = "all"
```

https://play.golang.org/p/z3R1RW91dVA